### ISSUE TITLE
Docker Installation Instructions Don't Work

### ISSUE THREAD

**Issue by lack-of-gravitas** [Sylius/Sylius#14260]
**Sylius version affected**: 1.12

**Description**  
Docker instructions do not work. Containers are started but localhost receives 502 Bad Gateway and localhost:8000 does not exist. MailHog works but nothing else does.

**Steps to reproduce**  
0 - instructions here
https://docs.sylius.com/en/1.12/book/installation/installation_with_docker.html
1 - git clone https://github.com/Sylius/Sylius.git sylius
2 - cd sylius
3 - docker-compose up -d (also attempted docker compose up -d)
3 - docker-compose up -d

**Actual Result**
- containers indicated started
- cannot access admin at localhost

**Expected Result**
- sylius admin accessible at localhost
![image](https://user-images.githubusercontent.com/10094964/186376087-9d7fc071-123c-4708-95e0-e8d66afc3d38.png)

**Comment by coldic3** [comment]
Hey @lack-of-gravitas,

What's your operating system?
Additionally, could you paste the results of both `docker-compose logs app` and `docker-compose logs mysql`?

**Comment by AymanGoT** [comment]
@coldic3 , 
I have the same issue
i followed the installation and continued with docker and i just can access port 35729 and db but the port 9000 not accessible.

there is error in the node which say that port 35729 is used ! 
any idea what might be the issue ?

```
ylius-platform-node-1     | yarn run v1.22.19
sylius-platform-node-1     | $ gulp watch
sylius-platform-node-1     | [08:20:09] Requiring external module babel-register
sylius-platform-node-1     | [08:20:09] Using gulpfile /srv/sylius/gulpfile.babel.js
sylius-platform-node-1     | [08:20:09] Starting 'watch'...
sylius-platform-node-1     | [08:20:09] Starting 'watchAdmin'...
sylius-platform-node-1     | [08:20:09] Starting 'watchShop'...
sylius-platform-node-1     | [gulp-chug] Gulpfile, gulpfile.babel.js, contents is empty. Reading directly from disk...
sylius-platform-node-1     | [gulp-chug] Spawning process /srv/sylius/node_modules/gulp/bin/gulp.js with args /srv/sylius/node_modules/gulp/bin/gulp.js --gulpfile gulpfile.babel.js watch --rootPath ../../../../../../../public/assets --nodeModulesPath ../../../../../../../node_modules from directory /srv/sylius/vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle...
sylius-platform-node-1     | [gulp-chug] Gulpfile, gulpfile.babel.js, contents is empty. Reading directly from disk...
sylius-platform-node-1     | [gulp-chug] Spawning process /srv/sylius/node_modules/gulp/bin/gulp.js with args /srv/sylius/node_modules/gulp/bin/gulp.js --gulpfile gulpfile.babel.js watch --rootPath ../../../../../../../public/assets --nodeModulesPath ../../../../../../../node_modules from directory /srv/sylius/vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:10] [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) Requiring external module babel-register
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:10] [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) Requiring external module babel-register
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:10] Using gulpfile /srv/sylius/vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:10] Starting 'watch'...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:10] Starting 'watchAdmin'...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:10] [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) Starting 'buildAdminJs'...
sylius-platform-node-1     | [08:20:10] Starting 'buildAdminCss'...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:10] [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) Starting 'buildAdminImg'...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:10] Using gulpfile /srv/sylius/vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:10] [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) Starting 'watch'...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:10] Starting 'watchShop'...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:10] [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) Starting 'buildShopJs'...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:10] Starting 'buildShopCss'...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:10] Starting 'buildShopImg'...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) 
sylius-platform-node-1     | ... Uhoh. Got error listen EADDRINUSE: address already in use :::35729 ...
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) Error: listen EADDRINUSE: address already in use :::35729
sylius-platform-node-1     |     at Server.setupListenHandle [as _listen2] (net.js:1331:16)
sylius-platform-node-1     |     at listenInCluster (net.js:1379:12)
sylius-platform-node-1     |     at Server.listen (net.js:1465:7)
sylius-platform-node-1     |     at Server.listen (/srv/sylius/node_modules/tiny-lr/src/server.js:264:19)
sylius-platform-node-1     |     at Function.exports.listen (/srv/sylius/node_modules/gulp-livereload/index.js:97:18)
sylius-platform-node-1     |     at watchShop (/srv/sylius/vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js:239:14)
sylius-platform-node-1     |     at bound (domain.js:421:15)
sylius-platform-node-1     |     at runBound (domain.js:432:12)
sylius-platform-node-1     |     at asyncRunner (/srv/sylius/node_modules/async-done/index.js:55:18)
sylius-platform-node-1     |     at processTicksAndRejections (internal/process/task_queues.js:77:11)
sylius-platform-node-1     | 
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) You already have a server listening on 35729
sylius-platform-node-1     | You should stop it and try again.
sylius-platform-node-1     | 
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:11] /srv/sylius/public/assets/shop/css/style.css.map reloaded.
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:12] [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) /srv/sylius/public/assets/shop/css/style.css reloaded.
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:13] /srv/sylius/public/assets/admin/css/style.css.map reloaded.
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:13] /srv/sylius/public/assets/admin/css/style.css reloaded.
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:13] [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) Finished 'buildShopJs' after 3.01 s
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:13] Finished 'buildShopImg' after 3.02 s
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/ShopBundle/gulpfile.babel.js) [08:20:13] Finished 'buildShopCss' after 3.04 s
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:14] Finished 'buildAdminImg' after 3.98 s
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:14] Finished 'buildAdminJs' after 3.99 s
sylius-platform-node-1     | [gulp-chug] (vendor/sylius/sylius/src/Sylius/Bundle/AdminBundle/gulpfile.babel.js) [08:20:14] Finished 'buildAdminCss' after 4 s

```
and this is the result when checking the processes on port 35729 on my machine just in case

```
sudo ss -tulpn | grep :35729
tcp    LISTEN  0       1024                0.0.0.0:35729          0.0.0.0:*      users:(("docker-proxy",pid=363892,fd=4))                                       
tcp    LISTEN  0       1024                   [::]:35729             [::]:*      users:(("docker-proxy",pid=363898,fd=4))                                       
```

**Comment by AymanGoT** [comment]
```
and this is the result when checking the processes on port 3527
```

**Comment by elboletaire** [comment]
I cannot even get sylius installed using the given docker-compose file. In my case there are problems both in the dockerfile and the docker-compose. I'm running docker from an arch distro under WSL (windows); but this should not be a problem since we're using docker.

The Dockerfile fails because curl is not installed and is required in a following step. Just adding curl does not work because then it fails with the installation, so it needs `-y curl` before `php8.0-xdebug`.

After adding those and properly building the container image, then the app fails with "The server ID parameter is not set; please run 'blackfire agent:config' to configure it." 

![imatge](https://user-images.githubusercontent.com/153305/189855019-04820891-231b-42ef-9148-1746f6f6c103.png)
(I later realize this is an error of the blackfire container)

So I add a custom container with a `command: blackfire agent:config`. But then I get an error because composer has not run `composer install` thus not adding the vendor autoload files.

So I change the command to `composer install`, which runs ok. Then I remove the added command to the docker-compose (I know this can be done by adding a different container, but I'm just trying things out), and run `up -d` again.

When trying to access via a web browser, I realize the `sylius_dev` database is not created... so I get back to add a custom command, this time with `bin/console sylius:install --no-interaction`.

After doing all this, I can access my store, but the styles are not installed (I guess everything related to yarn/gulp).

Edit: I accidentally sent this message before finishing with my final docker-compose. Here's the working docker-compose file (~~without the gulp/yarn stuff, which I'm currently trying to fix~~ fixed already):

~~~yaml
services:
    installer:
        build:
            context: .docker/dev
        environment:
            APP_ENV: "dev"
            DATABASE_URL: "mysql://root:mysql@mysql/sylius_%kernel.environment%?charset=utf8mb4"
#            DATABASE_URL: "pgsql://root:postgres@postgres/sylius_%kernel.environment%?charset=utf8" # When using postgres
            PHP_DATE_TIMEZONE: "Europe/Warsaw"
            PHP_XDEBUG_IDEKEY: "PHPSTORM"
            MAILER_URL: "smtp://mailhog:1025"
        volumes:
            - ./:/app:delegated
            - ./.docker/dev/php.ini:/etc/php/8.0/fpm/php.ini:delegated
            - ./.docker/dev/php.ini:/etc/php/8.0/cli/php.ini:delegated
        depends_on:
            mysql:
                condition: service_healthy
        networks:
            - sylius
        command: bash -c "composer install && bin/console sylius:install --no-interaction && yarn && yarn gulp"

    app:
        build:
            context: .docker/dev
        environment:
            APP_ENV: "dev"
            DATABASE_URL: "mysql://root:mysql@mysql/sylius_%kernel.environment%?charset=utf8mb4"
#            DATABASE_URL: "pgsql://root:postgres@postgres/sylius_%kernel.environment%?charset=utf8" # When using postgres
            PHP_DATE_TIMEZONE: "Europe/Warsaw"
            PHP_XDEBUG_IDEKEY: "PHPSTORM"
            MAILER_URL: "smtp://mailhog:1025"
        volumes:
            - ./:/app:delegated
            - ./.docker/dev/php.ini:/etc/php/8.0/fpm/php.ini:delegated
            - ./.docker/dev/php.ini:/etc/php/8.0/cli/php.ini:delegated
        ports:
            - 8080:80
        depends_on:
            installer:
                condition: service_completed_successfully
            mysql:
                condition: service_healthy
        networks:
            - sylius

    mysql:
        image: mysql:5.7
        platform: linux/amd64
        environment:
            MYSQL_ROOT_PASSWORD: mysql
            MYSQL_DATABASE: sylius_dev # this is required because, apparently, sylius:install does not do it for us
        networks:
            - sylius
        healthcheck:
            test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
            timeout: 20s
            retries: 10

#    postgres:
#        image: postgres:13-alpine
#        environment:
#            POSTGRES_USER: root
#            POSTGRES_PASSWORD: postgres
#        ports:
#            - 5432:5432
#        networks:
#            - sylius

    mailhog:
        image: mailhog/mailhog
        ports:
            - 8025:8025 # UI
        networks:
            - sylius

    blackfire:
        image: blackfire/blackfire:2
        environment:
            BLACKFIRE_LOG_LEVEL: 4
            BLACKFIRE_SERVER_ID: ~
            BLACKFIRE_SERVER_TOKEN: ~
            BLACKFIRE_CLIENT_ID: ~
            BLACKFIRE_CLIENT_TOKEN: ~
            BLACKFIRE_DISABLE_LEGACY_PORT: 1
        networks:
            - sylius

networks:
    sylius:
        driver: bridge
~~~
Edit2: ~~I'm updating the docker-compose as I progress. I'm actively working on the yarn/gulp stuff.~~ Updated with yarn stuff

**Comment by coldic3** [comment]
Just to clarify because I'm afraid we can mix contexts here – if you are going to run your app with Sylius you should use docker-compose from [Sylius/Sylius-Standard](https://github.com/Sylius/Sylius-Standard) as docker-compose in `Sylius/Sylius` is dedicated for contribution.

Therefore as far as I know you run docker containers configured in `Sylius/Sylius` but they don't deliver you ready-to-work application – @Ferror please correct me if I'm wrong. You still have to execute all installation commands inside a container on your own and adjust the docker-compose to your needs in `docker-compose.override.yml` for example.

**Comment by Ferror** [comment]
@elboletaire @AymanGoT @lack-of-gravitas if you are developing a new project you have to first create a project from Sylius-Standard or clone the git repository `git clone git@github.com:Sylius/Sylius-Standard.git` then follow the installation instruction.

The Sylius/Sylius repository is the whole platform/framework for developing the Sylius you use in the business project.

**Comment by elboletaire** [comment]
In my defense, reading the documentation I cannot know that: https://docs.sylius.com/en/1.12/book/installation/installation_with_docker.html

The only place where that "standard" version is shown is when installing using composer: https://docs.sylius.com/en/1.12/getting-started-with-sylius/installation.html

So IMHO, the documentation should be updated to show this.

**Comment by elboletaire** [comment]
Sorry I deleted the previous message instead of editing it 🤦🏼 

I said that following the documentation I cannot know that.

Re-reading the docker documentation I can see it speaks about "sylius standard", but here in this thread is the first time I know the real difference between them two. Following the documentation I started [here](https://docs.sylius.com/en/1.12/getting-started-with-sylius/installation.html), then clicked on the docker link, and there it speaks about "sylius standard", but I don't see how should I know they're different just by following the installation process. Maybe a clarification note could be added in the installation section.

Edit: reading the documentation again from the "getting started" point, I don't see any place where you show the difference between the two packages: https://docs.sylius.com/en/1.12/getting-started-with-sylius/index.html.

First step is just an index, next step is the installation process where the word "standard" only appears in the composer command. If you go then to the docker installation (following a link in that same page), no mentions of standard at all (other than a reference to "sylius standard", but there's no way to know they're different).

Edit2: BTW, the docker-compose file coming in sylius-standard works for me (at least it's accessible), although it does not seem to be building the node stuff. Running `docker compose exec node yarn build` fixes it tho.
